_posts/en/newsletters/2023-10-11-newsletter.md

```diff
@@ -0,0 +1,189 @@
+      and constrained to their intended uses before the program is
+      executed, thus preventing type-related errors and ensuring
+      program reliability. No downsides were noted."
+  a0link="https://bitcoincore.reviews/28107#l-38"
+
+  q1="Rather than the new class types `Txid` and a `Wtxid` _inheriting_
+      from `uint256`, should they  _include_ (wrap) a `uint256`?
+      What are the tradeoffs?"
+  a1="Those classes could do that, but it would cause much more code
+      churn (many more source lines would need to be touched)."
+  a1link="https://bitcoincore.reviews/28107#l-43"
+
+  q2="Why is it better to enforce types at compile-time instead of at
+      run-time?"
+  a2="Developers discover errors quickly as they're coding, rather
+      than relying on writing extensive test suites to catch bugs in runtime
+      (and these tests may still miss some errors). However, testing would
+      still be useful since type safety won't prevent consistent use of the
+      wrong type of transaction ID (but consistently) in the first place."
+  a2link="https://bitcoincore.reviews/28107#l-67"
+
+  q3="Conceptually, when writing new code that requires referring to
+      transactions, when should you use `txid` and when should you use
+      `wtxid`? Can you point to any examples in the code where using one
+      instead of the other could be very bad?"
+  a3="In general, use of `wtxid` is preferred since it commits to the
+      entire transaction. An important exception is the `prevout`
+      reference from each input to the output (UTXO) it's spending.
+      It must be a `txid` to retain compatibility with pre-segwit nodes.
+      (Note that it's not possible for only upgraded segwit nodes to use
+      `wtxid` here, because this would change the `txid` of the spending
+      transaction.)
+      An example of where it's important to use one and not the other is
+      given [here][wtxid example]."
+  a3link="https://bitcoincore.reviews/28107#l-85"
+
+  q4="In which concrete way(s) could using `transaction_identifier` instead
+      of `uint256` help find existing bugs or prevent the introduction of
+      new ones? On the other hand, could this change introduce new bugs?"
+  a4="Without this PR, a function that takes a `uint256` argument (such as
+      a block ID hash) could be passed a `txid`.
+      With this PR, this causes a compile-time error."
+  a4link="https://bitcoincore.reviews/28107#l-128"
+
+  q5="The [`GenTxid`][GenTxid] class already exists. How does it
+      already enforce type correctness, and how is it different from the
+      approach in this PR?"
+  a5="This class includes a hash and a flag indicating if the hash is a
+      `wtxid` or a `txid`, so it's still a single type rather than two
+      distinct types. This allows type checking, but it must be explicitly
+      programmed, and, more importantly, can only be done at run-time,
+      not compile-time."
```

Comment:
Maybe also mention the use case of being able to pass both, but still specifying which type. https://bitcoincore.reviews/28107#l-176